docs/cloud/secret-management.md

```diff
@@ -0,0 +1,92 @@
+
+:::
+
+:::caution Experimental feature
+
+Secret management is an experimental feature of Feldera. Setting up secrets
+currently involves a number of manual steps, which will be integrated
+into an automated/improved workflow in the production release.
+
+:::
+
+Within certain connector configurations, it is possible to refer to secrets rather than inputting them as plain text. This prevents them from being visible in the connector configurations. We currently make use of [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/) for management.
+
+
+## Feldera connector secrets
+
+Referring to secrets is currently enabled only for the Feldera Kafka input and output connector configuration.
+
+1. **Create kubernetes secret:** create a secret named `example` within the `feldera` namespace, which has a single key `example` with as value `something` (base64 encoded: `c29tZXRoaW5n`).
+   We can do this by creating a file `example-secret.yaml`:
+   ```
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: example
+     namespace: feldera
+   type: Opaque
+   data:
+     example: c29tZXRoaW5n
+   ```
+
+   ... and subsequently apply it using: `kubectl apply -n feldera -f example-secret.yaml`
+
+2. **Use the secret:** now, when we start the pipeline with a connector configuration value of `${secret:example}`, it will be resolved using a mounted secret volume containing the secret we just defined.
+
+
+## Kafka Connect (Debezium) secrets
+
+When creating a Kafka Connect Debezium connector to for example MySQL, it is required to provide database credentials. We make use of the Kafka Connect [FileConfigProvider](https://docs.confluent.io/platform/current/connect/security.html#fileconfigprovider), which allows us to refer to secrets in a similar way as Feldera connector secrets (with a slightly different pattern).
+
+Because Kafka Connect is already running, it is not possible to mount new secrets without restarting. Instead, we patch an already existing mounted secret and await for it to be synchronized with the running instance.
+
+**The steps are as follows:**
+
+1. The secret used is `kafka-connect-databases-secret` in the `kafka-connect` namespace.
+   It is mounted at `/etc/secret-volume-kafka-connect-databases` on the Kafka Connect instance.
+   Check its presence by running:
+   ```
+   kubectl get secret -n kafka-connect kafka-connect-databases-secret
+   ```
+
+2. Suppose we want to store the MySQL database credentials. To do this, we need
+   to add the secret key `.database-123456.properties` with the following content:
+   ```
+   hostname=mysql.some.domain.example.com
+   port=3306
+   user=chosen_debezium_user_123456
+   password=chosen_debezium_password_123456
+   ```
+   ... which is base64 encoded:
```

Comment:
It would be good to show the bash command to encode the file.
`cat .database-123456.properties | base64`

Reply:
Good point. I've added base64 encoding instructions.